brd-ios/breadwallet/src/Redesign/Views/MainSwapView.swift

```diff
@@ -0,0 +1,201 @@
+// 
+//  MainSwapView.swift
+//  breadwallet
+//
+//  Created by Kenan Mamedoff on 05/07/2022.
+//  Copyright © 2022 Fabriik Exchange, LLC. All rights reserved.
+//
+//  See the LICENSE file at the project root for license information.
+//
+
+import UIKit
+import SnapKit
+
+struct MainSwapConfiguration: Configurable {
+    var shadow: ShadowConfiguration?
+    var background: BackgroundConfiguration?
+}
+
+struct MainSwapViewModel: ViewModel {
+    var fromFiatAmount: String
+    var fromCryptoAmount: String
+    var toFiatAmount: String
+    var toCryptoAmount: String
+}
+
+class MainSwapView: FEView<MainSwapConfiguration, MainSwapViewModel> {
+    private lazy var containerStackView: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .vertical
+        stack.distribution = .fill
+        return stack
+    }()
+    
+    private var topSwapCurrencyView = SwapCurrencyView()
+    private var bottomSwapCurrencyView = SwapCurrencyView()
+    
+    private lazy var dividerWithButtonView: UIView = {
```

Comment:
dont think im the biggest fan of composing views inside a lazy var...

either extract the logic to a separate view file.. or do all layouting inside setupSubviews